src/test/transaction_tests.cpp

```diff
@@ -238,28 +267,27 @@ BOOST_AUTO_TEST_CASE(tx_invalid)
             CTransaction tx(deserialize, stream);
 
             CValidationState state;
-            fValid = CheckTransaction(tx, state) && state.IsValid();
+            if (!CheckTransaction(tx, state) || !state.IsValid()) {
+                BOOST_CHECK_MESSAGE(test[2].get_str() == "BADTX", strTest);
+                continue;
+            }
 
             PrecomputedTransactionData txdata(tx);
-            for (unsigned int i = 0; i < tx.vin.size() && fValid; i++)
-            {
-                if (!mapprevOutScriptPubKeys.count(tx.vin[i].prevout))
-                {
-                    BOOST_ERROR("Bad test: " << strTest);
-                    break;
+            std::vector<unsigned int> verify_flags = ParseScriptFlags(test[2].get_str());
+            if (!CheckTx(tx, mapprevOutScriptPubKeys, mapprevOutValues, verify_flags[0], txdata, strTest, false))
+                BOOST_ERROR("Tx unexpectedly passed: " << strTest);
+            for (size_t f = 1; f < verify_flags.size(); f++) {
+                unsigned int flags = verify_flags[f];
+                if (!(flags & SCRIPT_VERIFY_CLEANSTACK)) {
+                    flags &= ~SCRIPT_VERIFY_WITNESS;
```

Comment:
Should the implicit conversion from negative to unsigned be made explicit here?

Reply:
Done. Also fixed a bug as I swapped SCRIPT_VERIFY_CLEANSTACK and SCRIPT_VERIFY_WITNESS here